engine/glfw/java/com/dynamo/android/DefoldActivity.java

```diff
@@ -464,4 +469,46 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     public static native void nativeOnActivityResult(Activity activity, int requestCode, int resultCode, Intent data);
+
+    /**
+     * Method to get device ids for any connected gamepads, joysticks etc
+     * Called from glfwAndroid.
+     * @return Array of device ids
+     */
+    public int[] getGameControllerDeviceIds() {
+        mGameControllerDeviceIds.clear();
+        for (int deviceId : InputDevice.getDeviceIds()) {
+            InputDevice device = InputDevice.getDevice(deviceId);
+            int sources = device.getSources();
+            // filter out only gamepads, joysticks and things which has a dpad
+            if (((sources & InputDevice.SOURCE_GAMEPAD) == InputDevice.SOURCE_GAMEPAD) ||
+                ((sources & InputDevice.SOURCE_JOYSTICK) == InputDevice.SOURCE_JOYSTICK) ||
+                ((sources & InputDevice.SOURCE_DPAD) == InputDevice.SOURCE_DPAD)) {
+                    mGameControllerDeviceIds.add(deviceId);
+            }
+        }
+
+        // create and copy device ids to an int[]
+        int[] deviceIds = new int[mGameControllerDeviceIds.size()];
```

Comment:
I'm not super happy with this allocation but I couldn't figure out another way to go from ArrayList<Integer> to int[]